### DISCUSSION TITLE
Correct installation for metals

### DISCUSSION THREAD

**Discussion by peter-jerry-ye** [scalameta/metals#4671] [Q&A] [answered]
Hi, I'm using Metals with Helix editor with a sbt project that is mixed with Scala and Java files. I had 

- coursier to install sbt and bloop and latest snapshot version of metals 
- System package manager that installed javac / java / java source files

The lsp seems to work and can give diagnostic information for scala files, but I still have some problems.

- In the `.metals/metals.log`, there are *huge amount* of errors all similar to this one:
> ERROR error processing file:///home/zihang/.cache/coursier/v1/https/repo1.maven.org/maven2/com/typesafe/config/1.4.2/config-1.4.2-sources.jarjava.nio.file.NoSuchFileException: /home/zihang/.cache/coursier/v1/https/repo1.maven.org/maven2/com/typesafe/config/1.4.2/config-1.4.2-sources.jar
- For the scala file, only the type signature of symbols are displayed, but not document content, nor does goto definition work for standard library.
- No type signature / autocompletion for Java files, but goto definition works.

So what else should I configure to make it work smoothly?

**Comment by tgodzik** [comment, marked as answer]
> In the .metals/metals.log, there are huge amount of errors all similar to this one:

Source jars are supposed to be downloaded when running build import. Any idea why it wouldn't be present in the coursier cache?

> For the scala file, only the type signature of symbols are displayed, but not document content, nor does goto definition work for standard library.

If the indexing fail, which seems to be the case here. We don't know where to find the definition or which scaladocs to present.

>No type signature / autocompletion for Java files, but goto definition works.

It's currently not supported in Java files.

**Reply by peter-jerry-ye** [reply]
> It's currently not supported in Java files.

Ok, seems I had some false memory.

> If the indexing fail, which seems to be the case here. We don't know where to find the definition or which scaladocs to present.

Yes, I found these information:
> 2022.11.23 21:43:20 WARN  Unable to run doctor. Make sure `isHttpEnabled` is set to `true`.
> 2022.11.23 21:43:22 INFO  no build target found for /home/zihang/scala/leetcode/src/main/scala/LeetCode.scala. Using presentation compiler with project's scala-library version: 3.2.1

I assume that on VSCode we need to trigger "import build". But how does that work on a standalone editor? Is the "import build" a lsp command? If that's the case, I may need to wait for the next version of helix when they support lsp command.

> Source jars are supposed to be downloaded when running build import. Any idea why it wouldn't be present in the coursier cache?

Not a slightest hint. It seems all the `*-source.jar` are missing, but this may be related to the previous problem that "import build" is not ran?

**Reply by tgodzik** [reply]
> I assume that on VSCode we need to trigger "import build". But how does that work on a standalone editor? Is the "import build" a lsp command? If that's the case, I may need to wait for the next version of helix when they support lsp command.

It is, but this can also be done via request message, when the build definition changes. You can run this also manually in the console. Just check the build tool page for that.

> Not a slightest hint. It seems all the *-source.jar are missing, but this may be related to the previous problem that "import build" is not ran?

most likely

**Reply by peter-jerry-ye** [reply]
Ok, I see, so it's [manual installation section](https://scalameta.org/metals/docs/build-tools/sbt#manual-installation) for sbt. I tried that and problem solved. Thank you very much for your help.

**Reply by ckipp01** [reply]
btw if you use the nightly version of helix you don't need to do the manual installation. Instead that version of helix has the `:lsp-workspace-command` which gives you all the workspace commands, one of them build build-import. Then you can just select it and get the import. Much nicer than needing to manually do anything.

**Reply by ckipp01** [reply]
Also in case you didn't know, we have a #helix-users channel on our scalameta discord. :) https://discord.gg/VxKvTSnF

**Reply by peter-jerry-ye** [reply]
> btw if you use the nightly version of helix you don't need to do the manual installation

Yes I noticed that. It's just I'm not a fan of building everything myself.

> Also in case you didn't know, we have a #helix-users channel on our scalameta discord.

That's nice. The future of helix+scala seems very bright. Thanks for the information.
